controllers/managedocs_controller.go

```diff
@@ -175,14 +175,23 @@ func (r *ManagedOCSReconciler) generateStorageCluster(
 	if reconcileStrategy == v1.ReconcileStrategyStrict {
 		// Get an instance of the desired state
 		desired := utils.ObjectFromTemplate(templates.StorageClusterTemplate, r.Scheme).(*ocsv1.StorageCluster)
-
 		if err := r.setStorageClusterParamsFromSecret(desired); err != nil {
 			return err
 		}
 		// Override storage cluster spec with desired spec from the template.
 		// We do not replace meta or status on purpose
 
-		sc.Spec = desired.Spec
+		if sc.Spec.StorageDeviceSets == nil {
+			sc.Spec = desired.Spec
+		} else {
+			if sc.Spec.StorageDeviceSets[0].Count <= desired.Spec.StorageDeviceSets[0].Count {
```

Comment:
It is a bad idea to select a device set by index, as it might change in future changes to the code. Always search an item by name.